arch/arc/core/isr_wrapper.S

```diff
@@ -202,6 +202,44 @@ From RIRQ:
  */
 
 SECTION_FUNC(TEXT, _isr_wrapper)
+#ifdef CONFIG_EXECUTION_BENCHMARKING
+	/* Temporary save R0 */
+	st r0, [sp]
```

Comment:
Well I guess we shouldn't act like that because we effectively rewrite a value stored in the tip of the stack. What we should do here is `push r0` but in case of enabled stack checking and SP already == STACK_TOP (`_ARC_V2_USTACK_TOP` or `_ARC_V2_S_USTACK_TOP` depending on availability and status of secure mode) we'll get an exception because `push r0` is:
```
SP = SP - 4
*SP = r0
```

That said if we're already reaching a STACK_TOP for a particular thread there's no point in trying to escape our fate and so it makes no sense in disabling stack checking just because of 3 registers we need to save: `r0`, `r1` and `blink`.

I'll do a re-spin now.